token-client/src/main/java/com/sap/cloud/security/xsuaa/client/DefaultOAuth2TokenKeyService.java

```diff
@@ -58,15 +68,15 @@ public String retrieveTokenKeys(URI tokenKeysEndpointUri, @Nullable String zoneI
 				LOGGER.debug("Received statusCode {}", statusCode);
 				String body = EntityUtils.toString(response.getEntity(), StandardCharsets.UTF_8);
 				if (statusCode != HttpStatus.SC_OK) {
-					throw OAuth2ServiceException.builder("Error retrieving token keys for x-zone_uuid " + zoneId)
+					throw OAuth2ServiceException.builder("Error retrieving token keys for x-app_tid " + tenantId)
 							.withUri(tokenKeysEndpointUri)
-							.withHeaders(X_ZONE_UUID + "=" + zoneId)
+							.withHeaders(X_APP_TID + "=" + tenantId)
```

Comment:
should we also log the value of clientId header?

Reply:
good point